tasks/hyrax-metadata-updates/tests/test-integration-index.js

```diff
@@ -312,3 +313,108 @@ test('Test generate path from ECHO-10 throws exception with broken config', asyn
     message: 'Provider not supplied in configuration. Unable to construct path'
   });
 });
+
+test.serial('Test updating ECHO10 metadata file in S3 when formats are present and correct', async (t) => {
+  // Set up mock Validation call to CMR
+  nock('https://cmr.earthdata.nasa.gov').post('/ingest/providers/GES_DISC/validate/granule/MOD11A1.A2017200.h19v04.006.2017201090724.cmr.xml')
+    .reply(200);
+  await setupS3(t, false, 'tests/data/echo10inWithFormat.xml');
+  const config = cloneDeep(event.config);
+  config.hyrax = {
+    formats: [
+      'HDF-5'
+    ]
+  };
+  const e = {
+    config: config,
+    input: t.context.payload.input
+  };
+
+  await hyraxMetadataUpdate(e);
+  // Verify the metadata has been updated at the S3 location
+  const metadataFile = t.context.payload.input.granules[0].files.find((f) => f.type === 'metadata');
+  const actual = await getS3Object(`${metadataFile.bucket}/${metadataFile.fileStagingDir}`, metadataFile.name);
+  const expected = fs.readFileSync('tests/data/echo10outWithFormat.xml', 'utf8');
+  t.is(actual.Body.toString(), expected);
+
+  await recursivelyDeleteS3Bucket(t.context.stagingBucket);
```

Comment:
If there are cleanup steps that you want to run after every test, it's just as good to add them to an `afterEach` hook. Otherwise, if you leave the cleanup code in the test, you should surround the test code in 333-338 in a `try/finally` block to ensure that the cleanup code gets run

Reply:
I'm not executing that after every test in this file though.

Reply:
OK, then you probably want to surround the block running the test code, here lines 333-338, in a `try` block and then run the cleanup in a `finally` for any test that you do want the cleanup. Otherwise, if `hyraxMetadataUpdate` threw an error because we broke something, then your cleanup would never happen

```
try {
  await hyraxMetadataUpdate(e);
  // Verify the metadata has been updated at the S3 location
  const metadataFile = t.context.payload.input.granules[0].files.find((f) => f.type === 'metadata');
  const actual = await getS3Object(`${metadataFile.bucket}/${metadataFile.fileStagingDir}`, metadataFile.name);
  const expected = fs.readFileSync('tests/data/echo10outWithFormat.xml', 'utf8');
  t.is(actual.Body.toString(), expected);
} finally {
  await recursivelyDeleteS3Bucket(t.context.stagingBucket);  
}
```

Reply:
Committed